packages/connect/src/SmoldotProvider/SmoldotProvider.ts

```diff
@@ -280,34 +281,33 @@ export class SmoldotProvider implements ProviderInterface {
           }
         }
       })
-      .then((client: smoldot.SmoldotClient) => {
-        this.#client = client;
-        this.#connectionStatePingerId = setInterval(
-          this.#checkClientPeercount, this.healthPingerInterval);
-      })
-      .catch((error: Error) => {
-        this.emit('error', error);
-        return Promise.reject(error);
-      });
+      this.#connectionStatePingerId = setInterval(
+      this.#checkClientPeercount, this.healthPingerInterval);
+    } catch(error: unknown) {
+      this.emit('error', error);
+    }
   }
 
   /**
    * @description Manually "disconnect" - drops the reference to the WASM client
    */
   // eslint-disable-next-line @typescript-eslint/require-await
   public async disconnect(): Promise<void> {
-    if (this.#client) {
-      this.#client = undefined;
-    }
-
-    if (this.#connectionStatePingerId !== null) {
-      clearInterval(this.#connectionStatePingerId);
-    }
+    try {
```

Comment:
This `try` / `catch` covers more than it needs to and could accidentally mask unreasoned about errors.  I think you meant only to wrap the `disconnect()` in a `try` / `catch`